### DISCUSSION TITLE
RunReactants gives extra negative sign in the product

### DISCUSSION THREAD

**Discussion by manajit-das** [rdkit/rdkit#6879] [Q&A] [answered]
I am trying AllChem.ReactionFromSmarts to get the products of a reaction. In the reaction template though there isn't negative sign at an oxygen atom (see the example)  in the product side
![rdkitHelp](https://github.com/rdkit/rdkit/assets/59758360/276af994-2f3b-46de-948e-86de5e1c516a)
, still it gives extra negative sign. Here is my code.
from rdkit.Chem import AllChem
mis_complex = AllChem.ReactionFromSmarts('[*:1][O-:2].[c:3]1([F:4])[c:5][c:6][c:7][c:8][c:9]1.[Na+:10]>>[*:1][O:2][C:3]1([F:4])[C:5]=[C:6][C:7]=[C:8][C-:9]1.[Na+:10]')
mis_complex

ps=mis_complex.RunReactants((Chem.MolFromSmiles('[O-]C1=CC=CC=C1'), Chem.MolFromSmiles('FC1=C([N+]([O-])=O)C=C([N+]([O-])=O)C=C1'), Chem.MolFromSmiles('[Na+:10]')))
ps[0][0]

I want to know what is wrong here, and how to avoid it?
Thanks.

**Comment by dehaenw** [comment, marked as answer]
You can specify neutral charge explicitly in the product side of your rxn SMARTS, using `+0`

```python
from rdkit.Chem import AllChem
mis_complex = AllChem.ReactionFromSmarts('[*:1][O-:2].[c:3]1([F:4])[c:5][c:6][c:7][c:8][c:9]1.[Na+:10]>>[*:1][O+0:2][C:3]1([F:4])[C:5]=[C:6][C:7]=[C:8][C-:9]1.[Na+:10]')
mis_complex

ps=mis_complex.RunReactants((Chem.MolFromSmiles('[O-]C1=CC=CC=C1'), Chem.MolFromSmiles('FC1=C([N+](=O)[O-])C=C([N+](=O)[O-])C=C1'), Chem.MolFromSmiles('[Na+:10]')))
ps[0][0]
```

gives
![image](https://github.com/rdkit/rdkit/assets/66372095/e1f23f80-4bdf-4bb5-85d1-6fd859f7829e)

**Reply by manajit-das** [reply]
Thanks for your answer, it is helpful. Still I would like to know is it the expected behavior if the charge is not assigned explicitly in the template. Thanks again.

**Reply by dehaenw** [reply]
As I understand it, in general the logical with these reactions is not to assume anything and leave sanitizing and corrections for chemical intuition to the user. One reason for this is performance for example if you want to execute a sequence of several reactions on many reactants, it wouldnt make sense to sanitize every step and do some kind of reaction mechanism charge accounting in every step.

**Reply by manajit-das** [reply]
Thank you for the clarification.
